### DISCUSSION TITLE
useContract gas limit

### DISCUSSION THREAD

**Discussion by cormacncheese** [wevm/wagmi#839] [Q&A] [answered]
I'm running into the following error when calling my contract instantiated with `useContract`:

```
Error: cannot estimate gas; transaction may fail or may require manual gas limit
````

How do you resolve this in `wagmi`?

**Comment by tmm** [comment]
Would recommend not using `useContract`. Instead either:

- Use a contract query/mutation hook, like [`useContractRead`](https://wagmi.sh/docs/hooks/useContractRead) or [`useContractWrite`](https://wagmi.sh/docs/hooks/useContractWrite)
- Create an ethers [`Contract`](https://docs.ethers.io/v5/api/contract/contract/#Contract) directly

**Comment by cormacncheese** [comment, marked as answer]
Using overrides resolved this issue

```
 overrides: {
      gasLimit: 21000,
 }
```

full `usePrepareContractWrite`:
```
 const { config  } = usePrepareContractWrite({
    addressOrName: collection?.address,
    contractInterface: ERC721Drop.abi,
    functionName: 'purchase',
    args: [mintCounter],
    overrides: {
      gasLimit: 21000,
    },
  })
```
